install.sh

```diff
@@ -33,18 +33,18 @@ fi
 echo "Gateway configuration:"
 
 # Try to get gateway ID from MAC address
-# First try eth0, if that does not exist, try wlan0 (for RPi Zero)
-GATEWAY_EUI_NIC="eth0"
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
-    GATEWAY_EUI_NIC="wlan0"
-fi
 
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
+# Get first non-loopback network device that is currently connected
+NICS=$(ip link 2>&1 | grep "state UP" | grep -v LOOPBACK | sed -E 's/^[0-9]+: ([0-9a-z]+): .*/\1/')
```

Comment:
you could use `ip -oneline link 2>&1` to make sure it's all in one line and the `LOOPBACK` grepping doesn't leave orphan lines

Reply:
maybe instead of `ip link 2>&1 | grep "state UP" | grep -v LOOPBACK`, use `ip -oneline link show up 2>&1 | grep -v LOOPBACK` and get rid of the language issue as well

Reply:
Ahh! I was looking for something like `show up` but somehow overlooked that option!